### ISSUE TITLE
asyncio.TimeoutError during API requests is not retried

### ISSUE THREAD

**Issue by psontag** [nolar/kopf#840]
### Long story short

In our Operator we frequently see error messages where an `asyncio.TimeoutError` did not get caught during an API request to the Kubernetes server. In some cases this causes the Operator to crash.

### Kopf version

1.34.0

### Kubernetes version

1.19

### Python version

3.9.6

### Logs

```none
Activity 'configure' succeeded.
Initial authentication has been initiated.
Initial authentication has finished.
Activity 'login' succeeded.
Resource observer has failed:
Traceback (most recent call last): File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/aiokits/aiotasks.py"
 line 69
 in guard await coro File "/usr/local/lib/python3.9/site-packages/kopf/_core/reactor/observation.py"
 line 113
 in resource_observer resources = await scanning.scan_resources(groups=group_filter
 settings=settings
 logger=logger) File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/scanning.py"
 line 31
 in scan_resources resources.update(await coro) File "/usr/local/lib/python3.9/asyncio/tasks.py"
 line 614
 in _wait_for_one return f.result() # May raise f.exception(). File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/scanning.py"
 line 68
 in _read_new_apis rsp = await api.get('/apis'
 settings=settings
 logger=logger) File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/api.py"
 line 111
 in get response = await request( File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/auth.py"
 line 45
 in wrapper return await fn(*args
 **kwargs
 context=context) File "/usr/local/lib/python3.9/site-packages/kopf/_cogs/clients/api.py"
 line 78
 in request response = await context.session.request( File "/usr/local/lib/python3.9/site-packages/aiohttp/client.py"
 line 619
 in _request break File "/usr/local/lib/python3.9/site-packages/aiohttp/helpers.py"
 line 656
 in __exit__ raise asyncio.TimeoutError from None asyncio.exceptions.TimeoutError
```

### Additional information

The issue does seem to happen for other API calls as well and not only for the initial scanning of resources.
It looks like the `asyncio.TimeoutError` is not caught when making an API request:

https://github.com/nolar/kopf/blob/fe56b5676132c001af8edc12e2490ba6543ebbc5/kopf/_cogs/clients/api.py#L87

Is this intentional? If not I would be happy to provide a PR to change this.

**Comment by nolar** [comment]
It depends on what was aiohttp doing to raise this timeout. My understanding is that if there are connection timeouts, which can be interpreted as temporary server-side issues, it should be `aiohttp.errors.ClientTimeoutError`, a descendant of `aiohttp.errors.ClientConnectionError`.

**Comment by nolar** [comment]
After checking the source of aiohttp, there is no `ClientTimeoutError` anymore, only `ServerTimeoutError`. Some extra searching leads to this answer:

* https://stackoverflow.com/a/54140958/857383

While there are some factual mistakes (not one but two places; maybe they were added later), it is principally correct: `ServerTimeoutError` happens only in some specific sub-operations of a request, but for not for the whole request.

Looking for the tests of aiohttp — https://github.com/aio-libs/aiohttp/blob/v3.7.4/tests/test_client_functional.py — these two errors are clearly distinguished: `aiohttp.ServerTimeoutError` is used only for two cases, while in all other cases the author of aiohttp expects `asyncio.TimeoutError` (e.g. for header reading, etc).

As such, we should do as aiohttp's author does: expect a generic asyncio timeout error for any form of timeouts, not the aiohttp-specific one.

**So, the answer is:** 

* no, it is not intentional; 
* yes, it would be good if you send a PR including asyncio's timeout to the list of expected temporary server-side/networking errors.

**Comment by psontag** [comment]
Hey @nolar  thanks for merging the fix so quickly. Do you plan to make a new release soon or do you still want to wait with it?
We are seeing this problem quite frequently at the moment and would like to test this version. But of course we can also build our own kopf wheel if needed.

**Comment by nolar** [comment]
I am planning the next release or an RC for once Python 3.10 is released (scheduled for today, 2021-10-04), plus when GitHub Actions support it (i.e. as it appears in https://github.com/actions/python-versions/releases).

**Comment by nolar** [comment]
@philipp-sontag-by Released as `kopf==1.35rc1` ([link](https://github.com/nolar/kopf/releases/tag/1.35rc1)) — can you please try? If it works, I will promote it to the release.

**Comment by psontag** [comment]
Thank you, I will update our staging environment.
